### PR TITLE
feat(Cutter): Add vtkCutter class

### PR DESCRIPTION
**Pull request by laurennlam**
I implemented this class some weeks ago. I'd like to clean it but not enough time and I think it could be a good feature. For now, it works well with triangle cells. This should be improved (later...)

### CODE REVIEW

**Comment by laurennlam** [comment]
WIP - not ready to merge

**Comment by jourdain** [comment]
Any progress on that?

**Comment by laurennlam** [comment]
Not for now. I'll let you know when it will be ready.

**Comment by agirault** [comment]
Hi @laurennlam 
Do you plan to do any more work on this? If not could you let me know what was left to look at?

I'd like to get something in vtk-js soon. I worked from your pr, rebased it on master and got a nice working example out of it:
https://github.com/Kitware/vtk-js/compare/master...agirault:AddingVtkCutter?diff=unified&expand=1&name=AddingVtkCutter

**Comment by laurennlam** [comment]
Hi @agirault 
For now I don't plan to work on it.
I don't remembered if all cell's types are managed by the vtkCutter. It has to be checked.

File: Sources/Filters/Core/Cutter/index.js [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+              points.getPoint(pointID2, x2);
+
+              // Compute the intersected point on edge
+              const computedIntersectedPoint = [];
+              for (let j = 0; j < 3; j++) {
+                computedIntersectedPoint[j] = x1[j] + (t * (x2[j] - x1[j]));
+              }
+
+              intersectedEdgesList.push({ pointEdge1: pointID1, // id of one point of the edge
+                point2Edge: pointID2, // id of one point of the edge
```
Comment (agirault) [inline Sources/Filters/Core/Cutter/index.js]: @laurennlam that was the source of the issue i showed you (should be pointEdge2 not point2Edge)

Comment (laurennlam) [reply Sources/Filters/Core/Cutter/index.js]: Thanks for the report @agirault and sorry for this basic error...

Comment (agirault) [reply Sources/Filters/Core/Cutter/index.js]: No problem at all! And no, I have a pr coming up with this and various performance improvements. Thank you!

**Comment by agirault** [comment]
Superseeded by https://github.com/Kitware/vtk-js/pull/655

**Pull request closed without merging**